docs/roadmap.md

```diff
@@ -0,0 +1,169 @@
+  * Disabled by default. Can be enabled with configuration option
+    `scalac.diagnostics.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+* [x] sbt server: sbt 1.1 added LSP support so that build errors are reported
+      directly to the editor. Metals can connect to sbt server as a client to
+      trigger batch compilation on file save. Guarantees diagnostics are always
+      in-sync with build diagnostics but at the cost of higher latency.
+  * Disabled by default. Can be enabled with configuration option
+    `sbt.enabled=true`.
+  * Requires sbt 1.1 or later.
+* [x] Linting with [Scalafix][].
+  * Requires [semanticdb-scalac](#semanticdb-scalac).
+  * Supports both built-in and custom linter rules.
+
+## Robust navigation
+
+Goto definition and find references is a great way to navigate and learn about a
+codebase. Metals uses [SemanticDB][], a data model for semantic information
+about programs in Scala and other languages, to power code navigation. Indexing
+happens with during batch compilation in the build tool, using a similar
+architecture as [Index-While-Building][] in XCode 9.
+
+* Goto definition (`textDocument/definition`). Requires
+  [semanticdb-scalac](#semanticdb-scalac).
+  * [x] project -> project.
+  * [x] project -> Scala dependency. Requires [`metalsSetup`](#metalssetup).
+  * [x] project -> Java dependency. Requires [`metalsSetup`](#metalssetup).
+  * [ ] dependency -> dependency.
+* [x] Find references (`textDocument/references`). Works the same as goto
+      definition.
+  * [x] project -> project.
+  * [ ] project -> Scala dependency.
+  * [ ] project -> Java dependency.
+  * [ ] dependency -> dependency.
+* [x] Highlight references to symbol at position
+      (`textDocument/documentHighlight`).
+* [x] Open symbol in file (`textDocument/documentSymbol`)
+* [x] Open symbol in workspace (`workspace/symbol`)
+* [x] Symbol outline in the sidebar as you type (`textDocument/documentSymbol`).
+* [ ] Goto implementation (`textDocument/implementation`)
+* [ ] Goto type definition (`textDocument/typeDefinition`)
+* [x] Show type of symbol at position (`textDocument/hover`). Requires
+      [semanticdb-scalac](#semanticdb-scalac).
+* [ ] Show type of expression at position (`textDocument/hover`).
+
+## Fast completions
+
+Auto completions are a great way to explore a library interface and save
+keystrokes while writing code. Completions are sensitive to latency since
+suggestions update on every keystroke. Correct and fast completions are
+difficult to support in Scala due to several challenges:
+
+1.  the presentation compiler does not lend itself well to interactive editing
+2.  some completions semantics are are difficult to reproduce outside of the
+    compiler
+    * extension methods depend on implicit conversions
+    * whitebox macros can refine result types which influences completion
+      suggestions
+
+It will take a while until Metals can robustly support completions.
```

Comment:
Why do we say this only in this section, but not in other, like in the PC one? I suggest we leave this sentence out.

Reply:
Removed, and added

> Given these challenges, we are still exploring how to best support correct and fast completions.